### PR TITLE
Implement Color matrix on select image

### PR DESCRIPTION
**Pull request by arslandevcrew**
- Implement brightness
- Implement hue
- Implement saturation

### CODE REVIEW

**Comment by arslandevcrew** [comment]
[Implement adjust features on Image](https://trello.com/c/eEwSn1Zk/13-implement-adjust-features-on-image)

File: example/lib/features/adjust/color_filter_example.dart
Comment (WaseemAbbas-DevCrew) [inline example/lib/features/adjust/color_filter_example.dart:1]: please do the changes in the example as well once we remove the imageColorFilter widget from the library. 

The main purpose is to make our library widget free as possible.

File: lib/features/adjust/color_filter_generator.dart
Comment (WaseemAbbas-DevCrew) [inline lib/features/adjust/color_filter_generator.dart:1]: add proper documentation for each method.

Comment (arslandevcrew) [reply lib/features/adjust/color_filter_generator.dart:1]: documentation added

File: example/lib/features/adjust/color_filter_example.dart
Comment (arslandevcrew) [reply example/lib/features/adjust/color_filter_example.dart:1]: imageColorFilter widget removed from the library

File: lib/features/adjust/image_filter.dart
[the remark this answers is not in the document]
Comment (arslandevcrew) [reply lib/features/adjust/image_filter.dart:1]: imageColorFilter widget removed from the widget, now in library used on colorFilter to change the color

File: lib/features/adjust/color_filter_gen_impl.dart [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+
+
+class _ColorFilterGenerator implements ColorFilterGenerator {
+  @override
+   List<double> hueAdjustMatrix({required double value}) {
+    value = value * pi;
+
+    if (value == 0) {
+      return [
```
[the remark this answers is not in the document]
Comment (arslandevcrew) [reply lib/features/adjust/color_filter_gen_impl.dart:11]: done

File: lib/features/adjust/color_filter_gen_impl.dart [hunk 2]
```diff
@@ -0,0 +1,179 @@
+part of image_crop;
+
+
+
+class _ColorFilterGenerator implements ColorFilterGenerator {
+  @override
+   List<double> hueAdjustMatrix({required double value}) {
```
[the remark this answers is not in the document]
Comment (arslandevcrew) [reply lib/features/adjust/color_filter_gen_impl.dart:7]: min max values are defined in the function documentation, all values are supported but for better results values re mentioned.

File: lib/features/adjust/color_filter_gen_impl.dart
[the remark this answers is not in the document]
Comment (arslandevcrew) [reply lib/features/adjust/color_filter_gen_impl.dart:1]: done

File: lib/image_crop.dart
[the remark this answers is not in the document]
Comment (arslandevcrew) [reply lib/image_crop.dart:1]: Changed to `fhoto_editor`

**Review by WaseemAbbas-DevCrew — APPROVED** [review]

**Pull request merged**